### DISCUSSION TITLE
DWC relationships

### DISCUSSION THREAD

**Discussion by dustymc** [ArctosDB/arctos#4737] [General]
Reference: https://github.com/tdwg/dwc/issues/324

Summary: https://dwc.tdwg.org/terms/#resourcerelationship is (possibly?) a better mechanism to share relationships than associatedOccurrences, which we currently use (as a big ugly concatenated string).

@tucotuco and friends, I could use some help here. I'm (very) open to radical ideas. I see two obstacles:

1. We don't catalog Occurrences. We make up an OccurrenceID for DWC, but that's not what we use for relationships (or anything else).
2. Arctos relationships are not limited to primary keys.

For (1), mapping from OccurrenceID will just introduce redundancy. Given A-->B and 2 Occurrences of A, we'll say

A(Occurrence1)-->B
A(Occurrence2)-->B

That seems a little wonky from here, but perhaps it's "correct" and not redundant at all from the perspective of someone who does deal in Occurrences.

(2) (our documentation is https://handbook.arctosdb.org/documentation/other-identifying-numbers.html#id-references) seems more problematic to me. Arctos includes at least:

* relationship to an unencumbered and DWC-published record in Arctos. I can get OccurrenceIDs for these, everything works as I think it probably should (even if it's a little redundant because of (1)).
* relationship to an encumbered or unpublished record in Arctos. I can get OccurrenceIDs for these, but they may not mean anything from the DWC end. A mammal collection holds a "same individual as" genetic sample derived from a cultural object, and GBIF (nor anyone else as far as I know) does not accept cultural objects so it's "unpublished."
* relationship to a linked record outside of Arctos. I cannot get OccurrenceIDs for these (they probably do something as strange as we do), and I can't know if they're in the "DWC ecosystem" or not. (https://arctos.database.museum/guid/MSB:Mamm:235237 links - more or less - to an AMNH record, for example).
* "fuzzy" relationships - "this ate collector number 1" is possible (perhaps even common) in Arctos, "collector number 1" could mean millions of things (not all of which necessarily carry the identifier in any predictable manner) in any given context.

So, I think I have two questions:

1. Do we need to be doing something with DWC:Organism (which also isn't what we catalog, but perhaps it's closer than Occurrences) or something similar, or is faking Occurrences and replicating data when necessary "correct"?
2. Can I use (mostly) URIs and the occasional "collector number 1" for relatedResourceID, or does that need to reference a "key" in my Occurrences?

If this is limited to OccurrenceID, I think we'll inevitably end up with two problematic situations.

1. It will be lossy - I just can't include things for which I can't find an OccurrenceID, and so I can't share all of what I know via DWC
2. It will contain things that aren't actually keys in some Occurrence file - I won't know that some Arctos record hasn't been picked up by the DWC Portals (or some of them) for some reason, so the pointer to that cultural object (or recently-encumbered record, record rejected for licensing reasons, etc., etc., etc.) will be a dead end.

@jhpoelen I think you're the most active user of these data, your input is most welcome.

**Comment by campmlc** [comment]
@jhpoelen
Using ResourceRelationship makes sense to me below, but I am concerned about the comment by @deepreef :
"ResourceRelationships might work well in the limited context of exchanging associated occurrences within Darwin Core Archives, but when you want to exchange in a more graph-like format, a ResourceRelationship object will quickly become very problematic and you are probably better off with associatedOccurrences."

ResourceRelationshipClass
--
Identifier | http://rs.tdwg.org/dwc/terms/ResourceRelationship
Definition | A relationship of one rdfs:Resource (http://www.w3.org/2000/01/rdf-schema#Resource) to another.
Comments | Resources can be thought of as identifiable records or instances of classes and may include, but need not be limited to dwc:Occurrence, dwc:Organism, dwc:MaterialSample, dwc:Event, dwc:Location, dwc:GeologicalContext, dwc:Identification, or dwc:Taxon.
Examples | An instance of an Organism is the mother of another instance of an Organism. A uniquely identified Occurrence represents the same Occurrence as another uniquely identified Occurrence. A MaterialSample is a subsample of another MaterialSample.

**Comment by deepreef** [comment]
@campmlc :  That wasn't my comment you're quoting; I'm not sure who said that, or why that would be. If `ResourceRelationship` is problematic for graph-like formats, I'd like to understand why.  Basically, `ResourceRelationship` is the closest thing DwC has to a triple-store, so I would have thought it would be more friendly to graph-like formats than a single text blob with unenforced structure would be.

**Comment by campmlc** [comment]
Apologies @deepreef, it was @nielsklazenga who made the comment- my fault for attempting to Github by phone. Perhaps @nielsklazenga can clarify?

**Comment by jhpoelen** [comment]
> Basically, ResourceRelationship is the closest thing DwC has to a triple-store, so I would have thought it would be more friendly to graph-like formats than a single text blob with unenforced structure would be.

Note that ResourceRelationship definitely moves more towards a RDF triple-like form if the verb (or ```relationshipOfResource```) gets an id (proposed ```relationshipOfResourceID```) along with it's current free text field (or text blob).  See https://github.com/tdwg/dwc/issues/186 and https://github.com/tdwg/dwc/issues/283 .

I'd argue that adding the ID for the verb / relationship will help to make better use of the resourcerelationship records. Please help nudge @tucotuco and others to include this proposal into DwC by adding comments or thumbs up on https://github.com/tdwg/dwc/issues/283 .

**Comment by jhpoelen** [comment]
@dustymc Hey Dusty! Just curious - how did you end up addressing this issue re: DwC ResourceRelationships ?

**Comment by dustymc** [comment]
https://github.com/ArctosDB/internal/issues/168

We are moving towards https://docs.google.com/document/d/1QpXwole_j32QZAg6ddqOrAB5OOdqVJKdoKKzz06CK-o/edit

**Comment by jhpoelen** [comment]
Interesting! As far as I know, this "model" is not specific and more like an intention - like a plan to make a plan. So, I don't see how the resource relation issue has been resolved by this. But, perhaps you know more than I do. Please share!

**Comment by dustymc** [comment]
>now more than I do

Unlikely! Moving to discussions.
